### ISSUE TITLE
version 1.2.2 not working

### ISSUE THREAD

**Issue by xrandomhobox** [Vazkii/CMPDL#70]
I am trying to use this to download project ozone 2 using the URL 
https://www.curseforge.com/minecraft/modpacks/all-the-mods

It has a error 403 and fails. here is the logs

~ Starting magical modpack download sequence ~
Input URL: https://www.curseforge.com/minecraft/modpacks/all-the-mods
Input Version: 
File URL: https://www.curseforge.com/minecraft/modpacks/all-the-mods/files/latest
Final URL: https://www.curseforge.com/minecraft/modpacks/all-the-mods/files/latest

Modpack filename is latest
CMPDL Temp Dir is C:\Users\d\.cmpdl_temp
Modpack temporary directory is C:\Users\d\.cmpdl_temp\latest
Downloading zip file latest.zip
Error: class java.io.IOException: Server returned HTTP response code: 403 for URL: https://www.curseforge.com/minecraft/modpacks/all-the-mods/files/latest
sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
java.net.URL.openStream(Unknown Source)
vazkii.cmpdl.CMPDL.downloadFileFromURL(CMPDL.java:369)
vazkii.cmpdl.CMPDL.setupModpackMetadata(CMPDL.java:167)
vazkii.cmpdl.CMPDL.downloadFromURL(CMPDL.java:99)
vazkii.cmpdl.OperatorThread.run(OperatorThread.java:19)

**Comment by stefnotch** [comment]
https://github.com/iTitus/CMPDL/releases/tag/v1.3.0 seems to currently work

**Comment by quietust** [comment]
Neither version appears to work right now - I tried downloading Project Ozone 3, and while 1.2.2 gave the above error, the above-linked 1.3.0 successfully downloaded the ZIP file but failed to extract it (and gave other errors if I tried to extract it to the directory it was trying to create).

**Comment by jimmyli97** [comment]
iTitus' version expects a temp directory in C:/Users/YOUR_PROFILE/.cmpdl/MOD_PACK_NAME 

If you create this directory with nothing in it then it will work, at least for me when downloading Project Ozone 2.

**Comment by ewe2** [comment]
Definitely doesn't work in linux::

~ Starting magical modpack download sequence ~
Input URL: https://www.curseforge.com/minecraft/modpacks/better-minecraft-modpack
Input Version: 3194303
File URL: https://www.curseforge.com/minecraft/modpacks/better-minecraft-modpack/files/download/3194303
Final URL: https://www.curseforge.com/minecraft/modpacks/better-minecraft-modpack/files/download/3194303

Modpack filename is 3194303
CMPDL Temp Dir is /home/ewe2/.cmpdl_temp
Modpack temporary directory is /home/ewe2/.cmpdl_temp/3194303
Downloading zip file 3194303.zip
Error: class java.io.IOException: Server returned HTTP response code: 403 for URL: https://www.curseforge.com/minecraft/modpacks/better-minecraft-modpack/files/download/3194303
java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1924)
java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
java.base/java.net.URL.openStream(URL.java:1140)
vazkii.cmpdl.CMPDL.downloadFileFromURL(CMPDL.java:369)
vazkii.cmpdl.CMPDL.setupModpackMetadata(CMPDL.java:167)
vazkii.cmpdl.CMPDL.downloadFromURL(CMPDL.java:99)
vazkii.cmpdl.OperatorThread.run(OperatorThread.java:19)

**Comment by iTitus** [comment]
With the Overwolf migration they probably changed their CDN once again.
MultiMC will work for now, if you download the modpack zip manually and use the import feature of MultiMC on that zip file.

**Comment by ewe2** [comment]
> With the Overwolf migration they probably changed their CDN once again.
> MultiMC will work for now, if you download the modpack zip manually and use the import feature of MultiMC on that zip file.

Yeah I generally fall back on MultiMC, GDLauncher has similar issues with this. I do like to try out alternatives and will keep watching!

**Comment by iTitus** [comment]
I just checked with my fork, and it works. Try it here: https://github.com/iTitus/CMPDL/releases/
See #69. This fork is unofficial.

**Comment by ewe2** [comment]
Just tried it out with a pack, that's all good! Time for a pull request I think!

On Sat, 13 Feb 2021 at 03:54, iTitus <notifications@github.com> wrote:
>
> I just checked with my fork, and it works. Try it here: https://github.com/iTitus/CMPDL/releases/
> See #69. This fork is unofficial.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.



-- 

I love deadlines. I love the whooshing noise they make as they go by.
-- Douglas Adams.

**Comment by rom4ster** [comment]
> I just checked with my fork, and it works. Try it here: https://github.com/iTitus/CMPDL/releases/ See #69. This fork is unofficial.

Your fork cant detect file systems. Either A. you cant do it at all, or B. You have no accounting for external drives on windows.

**Comment by iTitus** [comment]
My fork uses standard Java APIs, if they don't work try to update Java.

But that does not matter, I would advise to use a launcher like Prism to download packs now - that even works with mods that you can only download manually.

**Comment by rom4ster** [comment]
> My fork uses standard Java APIs, if they don't work try to update Java.
> 
> But that does not matter, I would advise to use a launcher like Prism to download packs now - that even works with mods that you can only download manually.

I was in java 17 which is new enough, I use polymc. Asshole cunts on multimc had all this crap, and removed it which is why I had to go through all this.
